src/Microsoft.Diagnostics.Monitoring.WebApi/EgressValidationAttribute.cs

```diff
@@ -58,17 +72,22 @@ public void OnActionExecuting(ActionExecutingContext context)
 
             bool egressProviderGiven = context.HttpContext.Request.Query.TryGetValue(EgressQuery, out value);
 
-            if (!egressProviderGiven || value.ToString().Equals(""))
+            if (!egressProviderGiven || StringValues.IsNullOrEmpty(value))
             {
-                if (_egressOutputOptions.EgressMode == EgressMode.HttpDisabled)
+                if (!_egressOutputOptions.IsHttpEgressEnabled)
                 {
-                    ILogger<EgressValidationActionFilter> logger = context.HttpContext.RequestServices
-                        .GetRequiredService<ILoggerFactory>()
-                        .CreateLogger<EgressValidationActionFilter>();
-
-                    logger.LogError(Strings.ErrorMessage_HttpEgressDisabled);
+                    throw new EgressDisabledException();
+                }
+            }
+        }
 
-                    throw new ArgumentException("Http Egress is currently disabled.");
+        public class EgressDisabledException : Exception
+        {
+            public override string Message
+            {
+                get
+                {
+                    return "Http Egress is currently disabled.";
```

Comment:
Should this also be using Strings.ErrorMessage_HttpEgressDisabled? I just noticed the discrepancy between this and what gets logged.